packages/engine/paima-runtime/src/runtime-loops.ts

```diff
@@ -57,94 +56,123 @@ async function runPresync(
   startBlockHeight: number,
   stopBlockHeight: number | null
 ): Promise<void> {
-  const maximumPresyncBlockheight = stopBlockHeight
-    ? Math.min(startBlockHeight, stopBlockHeight)
-    : startBlockHeight;
   let presyncBlockHeight = await getPresyncStartBlockheight(
     gameStateMachine,
     funnelFactory.getExtensions(),
-    startBlockHeight,
-    maximumPresyncBlockheight
+    startBlockHeight
   );
 
-  if (presyncBlockHeight > maximumPresyncBlockheight) {
-    // doLog(
-    //   `[paima-runtime] Skipping presync (presync block height: ${presyncBlockHeight}, maximum presync block height: ${startBlockHeight})`
-    // );
-    return;
+  if (!ENV.CARP_URL) {
+    delete presyncBlockHeight[Network.CARDANO];
   }
 
   if (run) {
     doLog('---------------------------\nBeginning Presync\n---------------------------');
   }
-  while (run && presyncBlockHeight <= maximumPresyncBlockheight) {
-    const upper = Math.min(presyncBlockHeight + stepSize - 1, maximumPresyncBlockheight);
-    if (upper > presyncBlockHeight) {
-      doLog(`p${presyncBlockHeight}-${upper}`);
+  while (run && Object.keys(presyncBlockHeight).length !== 0) {
+    const upper = Object.fromEntries(
+      Object.entries(presyncBlockHeight)
+        .filter(([_network, h]) => h >= 0)
+        .map(([network, h]) => [network, h + stepSize - 1])
+    );
+
+    if (upper[Network.EVM] > presyncBlockHeight[Network.EVM]) {
+      doLog(`${JSON.stringify(presyncBlockHeight)}-${JSON.stringify(upper)}`);
     } else {
-      doLog(`p${presyncBlockHeight}`);
+      doLog(`${JSON.stringify(presyncBlockHeight)}`);
     }
 
     // eslint-disable-next-line @typescript-eslint/no-loop-func
-    presyncBlockHeight = await tx<number>(gameStateMachine.getReadWriteDbConn(), async dbTx => {
+    let newHeight = tx(gameStateMachine.getReadWriteDbConn(), async dbTx => {
       const chainFunnel = await funnelFactory.generateFunnel(dbTx);
       return await runPresyncRound(
         gameStateMachine,
         chainFunnel,
         pollingPeriod,
-        presyncBlockHeight,
-        upper
+        Object.entries(presyncBlockHeight).map(([network, height]) => ({
+          network: Number(network),
+          from: height,
+          to: upper[network],
+        }))
       );
     });
+
+    presyncBlockHeight = await newHeight;
```

Comment:
Is there a reason the await is on 2nd line like this?

Reply:
there was a reason at some point before, but not anymore